plugins/azure/redisCache/minimumTlsVersion.js

```diff
@@ -0,0 +1,48 @@
+var async = require('async');
+var helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'Minimum TLS Version',
+    category: 'Redis Cache',
+    description: 'Ensures that Azure Cache for Redis is using the latest TLS version.',
+    more_info: 'TLS versions 1.0 and 1.1 are known to be susceptible to attacks, and to have other Common Vulnerabilities and Exposures (CVE) weaknesses.So there\'s an industry- wide push toward the exclusive use of Transport Layer Security(TLS) version 1.2 or later.',
+    recommended_action: 'Ensure that Azure cache for Redis is using the latest TLS version',
+    link: 'https://docs.microsoft.com/en-us/azure/azure-cache-for-redis/cache-remove-tls-10-11',
+    apis: ['redisCaches:listBySubscription'],
+
+    run: function(cache, settings, callback) {
+        const results = [];
+        const source = {};
+        const locations = helpers.locations(settings.govcloud);
+
+        async.each(locations.redisCaches, function(location, rcb) {
+            const caches = helpers.addSource(cache, source,
+                ['redisCaches', 'listBySubscription', location]);
+
+            if (!caches) return rcb();
+
+            if (caches.err || !caches.data) {
+                helpers.addResult(results, 3, 'Unable to query Redis Caches: ' + helpers.addError(caches), location);
+                return rcb();
+            }
+
+            if (!caches.data.length) {
+                helpers.addResult(results, 0, 'No Redis Caches found', location);
+                return rcb();
+            }
+
+            for (let cache of caches.data) {
+                if (cache.minimumTlsVersion && (cache.minimumTlsVersion === '1.0' || cache.minimumTlsVersion === '1.1')) {
+                    helpers.addResult(results, 2, 'Redis Cache is not using the latest TLS Version', location, cache.id);
+                } else {
```

Comment:
should we have a else if case for !cache.minimumTlsVersion and report an error saying unable to get the tls version

Reply:
@ali-imran7 verify this. I think if we don't specify any version, then it 1.0 by default and I guess it does not return version in that case

Reply:
Yeah @dipsubha06 I agree. we should have one. 

@AkhtarAmir you are right. API does not return anything in case of the default version set. So, I added a case to generate an error with the message "Redis Cache is using the default TLS Version" .